### PR TITLE
Change workarea for application containers to the application ID, if available.

### LINKED ISSUE #22058: Unnecessary application expansion on restart
**Describe the bug**  
In limited situations, application binaries will be expanded into the workarea cache after restarting the server even when the application has not changed. This will occur intermittently, and only if a configuration file has been updated before restarting the server. 

**Steps to Reproduce**  
Start a server with an application installed, stop the server, update the timestamp on server.xml, start the server. 

**Expected behavior**  
The server should avoid unnecessary application binary expansion. 

**Diagnostic information:**  
 - OpenLiberty Version: 22.0.0.6 (and likely most previous versions)

**Additional context**

### PR DESCRIPTION
**Pull request by tbitonti**
This is a proof of concept.  I've verified that the configuration ID is null if not explicitly assigned, and found the assignment point (in ApplicationStateMachineImpl; see the PR for details), and assign the configuration ID if it is available instead of the service PID as the workarea folder for the application cache folder.

In addition to the primary change, there is a step which re-assigns the application container to the expanded application folder which was also updated (two code locations).

### CODE REVIEW

**Comment by tbitonti** [comment]
#build

**Review by brenthdaniel — COMMENTED** [review]
Deletion of the cache directories also needs to be handled (eg ApplicationConfigurator.uninstallApp() , ApplicationConfigurator.readyForAppsToStop())

**Comment by tbitonti** [comment]
The last commit also updated the state machine to handle the ID based directories during cleanup.

**Comment by tbitonti** [comment]
#build

**Comment by tbitonti** [comment]
Overall, FATs ran with few failures:
```
Tests Failures Errors Time Taken Run Order		
no component
84975 1 4
```
	
However, there are a number of FAT test failures which seem most probably related to the update:

There is a group of failures from JCA which look the same:

```
Tests Failures Errors Time Taken Run Order		

com.ibm.ws.jca_fat_derbyra FAT tests
33 0 1
com.ibm.ws.jca.fat.derbyra.LoginModuleInStandaloneResourceAdapterTest
1 0 1 0s 1

com.ibm.ws.jca_fat_embeddedrar FAT tests
17 0 1
com.ibm.ws.jca.fat.embedded.app.EmbeddedJCATest
1 0 1 0s 1

com.ibm.ws.jca_fat_enterpriseApp FAT tests
12 0 1
com.ibm.ws.jca.fat.enterpriseApp.EnterpriseAppTest
1 0 1 0s 1

java.lang.Exception: Errors/warnings were found in server com.ibm.ws.jca.fat.derbyra.loginModuleInRA logs:
 <br>[8/9/22, 21:03:52:655 PDT] 00000040 com.ibm.ws.classloading.internal.AppClassLoader              E CWWKL0002E: The system could not read class [org.apache.derby.jdbc.EmbeddedDataSource40] as resource [org/apache/derby/jdbc/EmbeddedDataSource40.class].
 <br>[8/9/22, 21:03:52:726 PDT] 00000040 com.ibm.ws.classloading.internal.AppClassLoader              E CWWKL0002E: The system could not read class [org.apache.derby.iapi.store.access.xa.XAXactId] as resource [org/apache/derby/iapi/store/access/xa/XAXactId.class].
	at componenttest.topology.impl.LibertyServer.checkLogsForErrorsAndWarnings(LibertyServer.java:3003)
	at componenttest.topology.impl.LibertyServer.stopServer(LibertyServer.java:2862)
```

There is an additional undetermined failure from microprofile which also looks related to the update.  This may a test error.
```
io.openliberty.microprofile.openapi.2.0.internal_fat FAT tests
26 1 0
io.openliberty.microprofile.openapi20.fat.deployments.DeploymentTest
testEar_EE9_FEATURES_MicroProfile_50_MpOpenApi_31
1 1 0 5s 1

junit.framework.AssertionFailedError: 2022-08-10-08:14:25:365 
Expected: is an empty collection
     got: <[[8/10/22, 8:14:23:901 UTC] 0000002c id=00000000 liberty.microprofile.openapi20.internal.ApplicationProcessor 3 Using OpenAPI model loaded from cache]>
	at io.openliberty.microprofile.openapi20.fat.deployments.DeploymentTest.assertCache(DeploymentTest.java:331)
	at io.openliberty.microprofile.openapi20.fat.deployments.DeploymentTest.testEar(DeploymentTest.java:143)
```

**Comment by tbitonti** [comment]
The test failures were from the first, incompletely coded, build.
The second build sees only two failures:
```
io.openliberty.microprofile.openapi20.fat.deployments.
DeploymentTest.testEar_EE9_FEATURES_MicroProfile_50_MpOpenApi_31
```
That test was coded to work around the cache location bug!  The test code is being updated.
```
com.ibm.ws.jmx.connector.server.rest_bvt bvt tests
com.ibm.ws.jmx.connector.bvt.JMXConnectorBVTTest
test_fileservice_getmetadata_longurl_readerRole_POST

java.net.SocketException: Socket closed
	at java.base/java.net.SocketInputStream.read(SocketInputStream.java:183)
```
That looks unrelated to the update.

Net: The second build is usable for testing.

**Comment by tbitonti** [comment]
#build

File: dev/com.ibm.ws.app.manager.war/src/com/ibm/ws/app/manager/ear/internal/EARDeployedAppInfoFactoryImpl.java [hunk 1]
```diff
@@ -40,18 +40,18 @@
 import com.ibm.wsspi.adaptable.module.InterpretedContainer;
 import com.ibm.wsspi.adaptable.module.UnableToAdaptException;
 import com.ibm.wsspi.adaptable.module.NonPersistentCache;
+import com.ibm.wsspi.adaptable.module.UnableToAdaptException;
```
Comment (jhanders34) [inline dev/com.ibm.ws.app.manager.war/src/com/ibm/ws/app/manager/ear/internal/EARDeployedAppInfoFactoryImpl.java:44]: You shouldn't duplicate the exception import.

File: dev/com.ibm.ws.app.manager/src/com/ibm/ws/app/manager/internal/statemachine/DownloadFileAction.java [hunk 2]
```diff
@@ -102,7 +101,7 @@ public void run() {
             output = new BufferedOutputStream(new FileOutputStream(downloadedFile));
 
             input = urlConection.getInputStream();
-            buf = new byte[1024]; //you can change the 1024 to a different size value
+            buf = new byte[32 * 1024]; //you can change the 1024 to a different size value
```
Comment (jhanders34) [inline dev/com.ibm.ws.app.manager/src/com/ibm/ws/app/manager/internal/statemachine/DownloadFileAction.java]: This change is out of left field it seems.  What is the reason for it?

File: dev/io.openliberty.microprofile.openapi.2.0.internal_fat/fat/src/io/openliberty/microprofile/openapi20/fat/deployments/DeploymentTest.java~ [hunk 3]
```diff
@@ -0,0 +1,397 @@
+/*******************************************************************************
```
Comment (jhanders34) [inline dev/io.openliberty.microprofile.openapi.2.0.internal_fat/fat/src/io/openliberty/microprofile/openapi20/fat/deployments/DeploymentTest.java~:1]: This looks like an unintended file in this PR.

**Review by jhanders34 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**